speeduino/storage.ino

```diff
@@ -11,6 +11,12 @@ A full copy of the license may be found in the projects root directory
 #include EEPROM_LIB_H //This is defined in the board .h files
 #include "storage.h"
 
+void startstorage(){
```

Comment:
Need to initialize the SPI flash. This function can be moved further back in the file to be with the rest of the helper functions

Reply:
Move it to board specific files, I had done this for FRAM

Reply:
Storage init needs to be called before board_init is called. Also the begin() function is part of the EEPROM instance. Best to keep all the EEPROM function calls in the storage.ino.

Reply:
I agree, but we need to keep ever board related code on their specifc files, see https://github.com/noisymime/speeduino/blob/f185b1e200b78d74f43d187cbef540a78ebf40b0/speeduino/board_stm32_generic.h#L27

With that line the FRAM lib is started.

Reply:
The downside of this is that we get calls/definitions to EEPROM in multiple places (and not only in storage where you would expect it)

I will take a look if it is possible to call a function in that location. Your example is a declaration and definition (or instantiation) of the variable EEPROM. That is not the same as function call to EEPROM.begin(). I will take a look when i get home tonight how to do this.

Reply:
Hmmm i did not get it to work without the call to EEPROM.begin() just before reading the configuration. 

If i try to use the SPI class instance during class instantiation of SPIAsEEPROM it does not send/receive any thing over the SPI (As seen with logic analyzer). Therefor this seems the only way for now.